app.py

```
@@ -927,6 +933,57 @@ def run_fake_task(task_name: str, cancel_event: threading.Event):
                 fake_state.append_task_log(task_name, f"HDSentinel unavailable: {hdsentinel_snapshot['error']}")
         elif task_name == 'Cloud Backup':
             run_fake_cloud_backup(cancel_event)
+        elif task_name == 'DDNS Update':
+            if cancel_event.is_set():
+                raise RuntimeError('Task was cancelled.')
+            ddns_script = runtime.repo_root / 'scripts' / 'ddns_update.py'
+
+            # Start the process non-blocking to allow cancellation
+            try:
+                proc = subprocess.Popen(
```

Ruff (0.15.9)

[error] `subprocess` call: check for execution of untrusted input

(S603)